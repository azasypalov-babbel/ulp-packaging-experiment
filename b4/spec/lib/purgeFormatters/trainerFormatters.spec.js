import {
  removeCorrectItems,
  makeCorrectItemsNonInteractive,
  makeUnattemptedItemsNonInteractive
} from '../../../src/lib/purgeFormatters/trainerFormatters';

const trainer = {
  item_groups: [
    {
      items: [
        {
          id: '012',
          attempt: { mistakes: 0 }
        },
        {
          id: '022',
          attempt: { mistakes: 1 }
        },
        {
          id: '042'
        },
        {
          id: '052',
          attempt: { mistakes: 0 }
        },
        {
          id: '062',
          attempt: { skipped: true }
        }
      ]
    },
    {
      items: [
        {
          id: '012',
          attempt: { mistakes: 0 }
        },
        {
          id: '022',
          attempt: { mistakes: 2 }
        },
        {
          id: '072'
        }
      ]
    },
    {
      items: [
        {
          id: '012',
          attempt: { mistakes: 0 }
        }
      ]
    }
  ]
};

describe('trainerFormatters', () => {
  describe('#removeCorrectItems', () => {
    test('works', () => {
      expect(removeCorrectItems(trainer)).toMatchSnapshot();
    });
  });

  describe('#makeCorrectItemsNonInteractive', () => {
    test('works', () => {
      expect(makeCorrectItemsNonInteractive(trainer)).toMatchSnapshot();
    });
  });

  describe('#makeUnattemptedItemsNonInteractive', () => {
    test('works', () => {
      expect(makeUnattemptedItemsNonInteractive(trainer)).toMatchSnapshot();
    });
  });
});
