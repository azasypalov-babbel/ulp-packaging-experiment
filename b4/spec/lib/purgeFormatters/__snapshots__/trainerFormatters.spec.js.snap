// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trainerFormatters #makeCorrectItemsNonInteractive works 1`] = `
Object {
  "item_groups": Array [
    Object {
      "items": Array [
        Object {
          "attempt": Object {
            "mistakes": 0,
          },
          "id": "012",
          "nonInteractive": true,
        },
        Object {
          "attempt": Object {
            "mistakes": 1,
          },
          "id": "022",
        },
        Object {
          "id": "042",
        },
        Object {
          "attempt": Object {
            "mistakes": 0,
          },
          "id": "052",
          "nonInteractive": true,
        },
        Object {
          "attempt": Object {
            "skipped": true,
          },
          "id": "062",
        },
      ],
    },
    Object {
      "items": Array [
        Object {
          "attempt": Object {
            "mistakes": 0,
          },
          "id": "012",
          "nonInteractive": true,
        },
        Object {
          "attempt": Object {
            "mistakes": 2,
          },
          "id": "022",
        },
        Object {
          "id": "072",
        },
      ],
    },
    Object {
      "items": Array [
        Object {
          "attempt": Object {
            "mistakes": 0,
          },
          "id": "012",
          "nonInteractive": true,
        },
      ],
    },
  ],
}
`;

exports[`trainerFormatters #makeUnattemptedItemsNonInteractive works 1`] = `
Object {
  "item_groups": Array [
    Object {
      "items": Array [
        Object {
          "attempt": Object {
            "mistakes": 0,
          },
          "id": "012",
        },
        Object {
          "attempt": Object {
            "mistakes": 1,
          },
          "id": "022",
        },
        Object {
          "id": "042",
          "nonInteractive": true,
        },
        Object {
          "attempt": Object {
            "mistakes": 0,
          },
          "id": "052",
        },
        Object {
          "attempt": Object {
            "skipped": true,
          },
          "id": "062",
        },
      ],
    },
    Object {
      "items": Array [
        Object {
          "attempt": Object {
            "mistakes": 0,
          },
          "id": "012",
        },
        Object {
          "attempt": Object {
            "mistakes": 2,
          },
          "id": "022",
        },
        Object {
          "id": "072",
          "nonInteractive": true,
        },
      ],
    },
    Object {
      "items": Array [
        Object {
          "attempt": Object {
            "mistakes": 0,
          },
          "id": "012",
        },
      ],
    },
  ],
}
`;

exports[`trainerFormatters #removeCorrectItems works 1`] = `
Object {
  "item_groups": Array [
    Object {
      "items": Array [
        Object {
          "attempt": Object {
            "mistakes": 1,
          },
          "id": "022",
        },
        Object {
          "id": "042",
        },
        Object {
          "attempt": Object {
            "skipped": true,
          },
          "id": "062",
        },
      ],
    },
    Object {
      "items": Array [
        Object {
          "attempt": Object {
            "mistakes": 2,
          },
          "id": "022",
        },
        Object {
          "id": "072",
        },
      ],
    },
  ],
}
`;
